File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Actions/Operation/ExecuteTask.cs
using Cen.Wms.Client.Actions.UI.Task;
using Cen.Wms.Client.Models.Enums;

namespace Cen.Wms.Client.Actions.Operation
{
    class ExecuteTask
    {
        public static void Run()
        {
            var taskType = TaskType.None;
            while (taskType != TaskType.Exit)
            {
                taskType = AskForTaskType.Run();

                if (taskType == TaskType.PurchaseByPapers)
                    PurchaseTaskAcceptByPapers.Run();

                if (taskType == TaskType.PurchaseByTask)
                    PurchaseTaskAcceptByTask.Run();
            }
        }
    }
}

File: TCDNew-master/src/ui.android/lib/actions/operation/ExecuteTask.dart
import 'package:tiq/actions/ui/task/AskForTaskType.dart';
import 'package:tiq/models/enums/TaskType.dart';
import 'package:flutter/material.dart';

import 'PurchaseTaskAcceptByPapers.dart';
import 'PurchaseTaskAcceptByTask.dart';

class ExecuteTask {
  Future<void> run(BuildContext context) async {
    var taskType = TaskType.none;
    while (taskType != TaskType.exit)
    {
      taskType = await AskForTaskType().run(context);

      if (taskType == TaskType.purchaseByPapers) {
        await PurchaseTaskAcceptByPapers().run(context);
      }

      if (taskType == TaskType.purchaseByTask) {
        await PurchaseTaskAcceptByTask().run(context);
      }
    }
  }
}